quintilesims/layer0
# Terraform: improve handling of sigterm

**What does this pull request do?**
Makes the user experience better when a user pressed `ctrl+c` during terraform execution; allows `WaitForDeployment()` and `WaitForJob()` to be cancelled without data loss. 

It also fixes a bug introduced by https://github.com/quintilesims/layer0/pull/300, where dockerruns aren't being diff'd correctly, resulting in deploys being re-created during every apply. 

**How should this be tested?**
Create a service using the terraform module, and press ctrl+c while it is in the "wait for deployment" state. The service should still exist in the terraform configuration - so the next `terraform apply` shouldn't try to re-create the service. 


**Checklist**


closes #295 
(reference the issue(s) this pull request closes)
